### ISSUE TITLE
Need a training dataset and  a separate validation dataset prepared in advance?

### ISSUE THREAD

**Issue by lxqpku** [microsoft/smart#13]
If I collect the pretraining data for one task with 1M timesteps, how can I divide the full replay buffer into training_dataset and validation dataset used below? Such as what the split ratio and any other details?
![屏幕截图 2023-11-30 215210](https://github.com/microsoft/smart/assets/45137511/fea9cc61-2528-4b4f-927b-7da724f4b30f)

**Comment by ycsun2017** [comment]
Hi @lxqpku, thank you for the question. If you would like to use a validation set, you can use the trajectories from a different replay buffer (collect with a different random seed). You can set it by train_replay_id and val_replay_id in the config. But since the downstream performance is evaluated by the environment score, the validation set does not play a big role here. We collect rollouts at the end of each epoch to see how well the policy is learned, as is common in reinforcement learning.
